core/runner.py

```diff
@@ -221,3 +225,43 @@ def _initialize_model(self, job):
                     error_message="",
                 )
         return results
+
+    def _average(self, job):
+        """
+        Average the weights in the job weighted by their omegas.
+        """
+        deserialized_new_weights = deserialize_weights(job.new_weights)
+        averaged_weights = self.weighted_running_avg(job.weights, deserialized_new_weights, job.sigma_omega, job.omega)
+        result = DMLResult(
+            status='successful',
+            job=job,
+            results={
+                'weights': averaged_weights,
+            },
+            error_message="",
+        )
+        return result
+    
+    def weighted_running_avg(self, sigma_x_i_div_w_i, x_n, sigma_w_i, w_n):
```

Comment:
Nitpick but documentation should be more clear on what the arguments are (i.e. explicitly state that the arguments are numpy arrays, what the other two arguments represent, etc.)

Reply:
Check latest commit